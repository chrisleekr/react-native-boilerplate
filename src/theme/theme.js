import {Dimensions} from 'react-native';
import {colors} from './colors';
const {width, height} = Dimensions.get('window');

// Themes:- Can alter values here. Can only be consumed through Context (see useTheme.js file)
const themes = {
  light: {
    name: 'light',
    barStyle: 'light-content',
    color: '#000000',
    labelColor: '#525E75',
    invertColor: '#ffffff',
    layoutBg: '#f0f0f0',
    headerBg: '#f6f7f8',
    cardBg: '#e4dfa7',
    accent: '#0071ff',
    error: '#B00020',
    errorText: '#ed4337',
    lightBg: '#F5EE9E',
    borderColor: '#B00020',
    buttonActiveColor: '#6200EE',
    buttonDisabledColor: 'rgba(0, 0, 0, 0.26)',
    card: {
      shadowColor: '#000000',
      borderColor: '#B00020'
    },
    ...colors
  },
  dark: {
    name: 'dark',
    barStyle: 'dark-content',
    color: '#ffffff',
    labelColor: '#F1DDBF',
    invertColor: '#000000',
    layoutBg: '#121212',
    headerBg: '#1e1e1e',
    cardBg: '#3f4045',
    accent: '#0071ff',
    error: '#B00020',
    errorText: '#ed4337',
    lightBg: '#F5EE9E',
    borderColor: '#ffffff',
    buttonActiveColor: '#BB86FC',
    buttonDisabledColor: 'rgba(255,255,255, 0.26)',
    card: {
      shadowColor: '#121212',
      borderColor: '#ffffff'
    },
    ...colors
  }
};

const window = {
  // app dimensions
  width,
  height
};

export {themes, window};
