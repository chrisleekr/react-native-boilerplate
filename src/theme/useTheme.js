/**
 * For Dark/Light Mode
 * @exports ThemeContext, ThemeProvider, useTheme
 */

import React, {useState, createContext, useContext} from 'react';
import {Appearance} from 'react-native';

// Import preconfigured themes from theme file
import {themes} from './theme';

// Context
export const ThemeContext = createContext(null);

// Provider to be used in index/App/or top of any parent
export const ThemeProvider = ({children}) => {
  const colorScheme = Appearance.getColorScheme();

  const [theme, setTheme] = useState(themes[colorScheme]);
  return (
    <ThemeContext.Provider value={{theme, setTheme}}>
      {children}
    </ThemeContext.Provider>
  );
};

// useTheme hook for easy access to theme and setTheme
export const useTheme = () => {
  const {theme, setTheme} = useContext(ThemeContext);

  const toggleTheme = v => {
    setTheme(v ? themes.dark : themes.light);
  };

  return {theme, toggleTheme};
};
