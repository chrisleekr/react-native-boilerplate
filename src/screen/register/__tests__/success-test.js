import React from 'react';
import {
  cleanup,
  render,
  waitFor,
  fireEvent
} from '@testing-library/react-native';

import {ThemeProvider} from '../../../theme/useTheme';

import SuccessScreen from '../success';

let rendered;

const mockNavigate = jest.fn();

const component = (
  <ThemeProvider>
    <SuccessScreen navigation={{navigate: mockNavigate}} />
  </ThemeProvider>
);

describe('SuccessScreen', () => {
  afterEach(cleanup);

  beforeEach(async () => {
    rendered = render(component);

    await waitFor(() => expect(rendered.toJSON()).not.toBeNull());
  });

  it('matches snapshot', () => {
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  describe('goBack', () => {
    beforeEach(async () => {
      fireEvent.press(rendered.getByTestId('back-button'));
    });

    it('triggers navigate', () => {
      expect(mockNavigate).toHaveBeenCalledWith('Login');
    });
  });
});
