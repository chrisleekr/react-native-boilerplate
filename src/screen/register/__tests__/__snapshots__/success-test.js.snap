// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessScreen matches snapshot 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "#f0f0f0",
      },
      Object {
        "flex": 1,
        "flexDirection": "column",
        "padding": 10,
      },
    ]
  }
  testID="register-success-screen"
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "left": 4,
        "opacity": 1,
        "position": "absolute",
        "top": 10,
      }
    }
    testID="back-button"
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 25,
          },
          undefined,
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ó±¦±
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": 50,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "aspectRatio": 1,
          },
          undefined,
          Object {
            "width": 50,
          },
        ]
      }
    >
      <View
        autoPlay={true}
        cacheComposition={true}
        collapsable={false}
        enableMergePathsAndroidForKitKatAndAbove={false}
        forwardedRef={[Function]}
        loop={true}
        onAnimationFinish={[Function]}
        onLayout={[Function]}
        progress={0}
        resizeMode="contain"
        sourceJson="{\\"v\\":\\"5.0.1\\",\\"fr\\":10,\\"ip\\":0,\\"op\\":200,\\"w\\":256,\\"h\\":256,\\"assets\\":[],\\"layers\\":[{\\"ind\\":1,\\"nm\\":\\"Layer 1\\",\\"ks\\":{\\"p\\":{\\"a\\":0,\\"k\\":[0,0]},\\"a\\":{\\"a\\":0,\\"k\\":[0,0,0]},\\"s\\":{\\"a\\":0,\\"k\\":[100,100,100]},\\"r\\":{\\"a\\":0,\\"k\\":0},\\"o\\":{\\"a\\":0,\\"k\\":100}},\\"ao\\":0,\\"ip\\":0,\\"op\\":200,\\"st\\":0,\\"bm\\":0,\\"sr\\":1,\\"ty\\":4,\\"shapes\\":[{\\"ty\\":\\"gr\\",\\"it\\":[{\\"ty\\":\\"sh\\",\\"d\\":1,\\"ks\\":{\\"a\\":0,\\"k\\":{\\"i\\":[[0,0],[0,-10.897],[20.864,-6.873],[1.152,-0.338],[-0.315,-1.58],[9.763,-5.619],[15.468,13.234],[1.598,1.533],[1.262,-1.097],[9.771,5.602],[-3.807,19.933],[-0.537,2.139],[1.627,0.55],[0,11.215],[-19.268,6.679],[-2.331,0.658],[0.397,2.009],[-9.485,5.459],[-16.436,-14.51],[-0.988,-0.942],[-1.545,1.34],[-9.494,-5.444],[4.434,-21.405],[0.327,-1.319],[-1.937,-0.661]],\\"o\\":[[19.657,6.728],[0,11.365],[-1.103,0.363],[0.411,1.644],[4.119,20.687],[-9.452,5.44],[-1.682,-1.442],[-1.269,1.219],[-15.969,13.879],[-9.461,-5.426],[0.416,-2.166],[-1.653,-0.47],[-20.07,-6.82],[0,-10.858],[2.154,-0.747],[-0.499,-1.986],[-3.992,-20.286],[9.892,-5.693],[1.022,0.904],[1.475,-1.416],[15.683,-13.569],[9.903,5.676],[-0.278,1.33],[1.968,0.563],[0,0]],\\"v\\":[[149.641,52.183],[182,80.347],[147.561,109.697],[144.176,110.747],[145.268,115.584],[136.452,158.893],[96.256,145.755],[91.336,141.292],[87.538,144.772],[45.399,158.807],[36.772,117.621],[38.202,111.161],[33.28,109.631],[0,80.347],[31.574,52.342],[38.31,50.235],[36.965,44.242],[45.267,2.285],[88.069,17.216],[91.085,19.985],[95.616,15.85],[136.32,2.049],[144.688,46.374],[143.782,50.347],[149.641,52.183]],\\"c\\":true},\\"hd\\":false}},{\\"ty\\":\\"sh\\",\\"d\\":1,\\"ks\\":{\\"a\\":0,\\"k\\":{\\"i\\":[[0,0],[0,0],[0,6.995],[16.843,5.764],[1.862,0.536],[3.631,-7.776],[-2.304,-7.25],[-0.994,0.328]],\\"o\\":[[0,0],[18.014,-5.935],[0,-6.567],[-1.741,-0.597],[-2.239,7.204],[3.804,7.964],[1.04,-0.307],[0,0]],\\"v\\":[[145.115,102.359],[145.116,102.359],[174.226,80.347],[147.11,59.493],[141.7,57.795],[132.842,80.386],[142.059,103.31],[145.115,102.359]],\\"c\\":true},\\"hd\\":false}},{\\"ty\\":\\"sh\\",\\"d\\":1,\\"ks\\":{\\"a\\":0,\\"k\\":{\\"i\\":[[0,0],[0.371,1.489],[8.799,-0.776],[5.129,-5.577],[-1.517,-1.297],[-5.735,3.302],[-1.084,5.809],[1.821,9.147]],\\"o\\":[[-0.285,-1.428],[-7.538,1.732],[-5.002,7.2],[1.532,1.465],[13.227,11.318],[2.696,-1.551],[1.207,-6.465],[0,0]],\\"v\\":[[137.641,117.086],[136.655,112.709],[112.006,116.486],[96.75,135.747],[101.327,139.895],[132.556,152.202],[138.515,140.969],[137.641,117.086]],\\"c\\":true},\\"hd\\":false}},{\\"ty\\":\\"sh\\",\\"d\\":1,\\"ks\\":{\\"a\\":0,\\"k\\":{\\"i\\":[[0,0],[0,0],[-7.062,6.138],[-1.129,1.087],[5.035,7.045],[7.455,1.639],[0.387,-2.015],[-5.741,-3.292],[-5.604,1.968]],\\"o\\":[[0,0],[6.236,-2.19],[1.182,-1.03],[-5.314,-5.671],[-8.82,-0.711],[-0.499,1.99],[-3.255,17.046],[2.698,1.546],[0,0]],\\"v\\":[[62.048,151.632],[62.048,151.632],[82.422,138.951],[85.889,135.776],[70.267,116.585],[45.74,113.055],[44.411,119.063],[49.283,152.11],[62.048,151.632]],\\"c\\":true},\\"hd\\":false}},{\\"ty\\":\\"sh\\",\\"d\\":1,\\"ks\\":{\\"a\\":0,\\"k\\":{\\"i\\":[[0,0],[-1.509,-0.429],[-3.652,7.904],[2.278,7.365],[2.068,-0.714],[0,-6.589],[-4.514,-3.844],[-8.875,-3.016]],\\"o\\":[[1.486,0.503],[2.299,-7.452],[-3.605,-7.787],[-2.107,0.592],[-16.479,5.713],[0,3.097],[5.023,4.277],[0,0]],\\"v\\":[[35.795,102.316],[40.288,103.714],[49.269,80.535],[40.398,57.682],[34.134,59.641],[7.774,80.347],[14.568,91.104],[35.795,102.316]],\\"c\\":true},\\"hd\\":false}},{\\"ty\\":\\"sh\\",\\"d\\":1,\\"ks\\":{\\"a\\":0,\\"k\\":{\\"i\\":[[0,0],[0,0],[-0.464,-1.848],[-8.56,0.789],[-5.287,5.645],[0.907,0.801],[6.089,-3.505],[-3.421,-17.381]],\\"o\\":[[0,0],[0.369,1.869],[7.533,-1.697],[4.991,-7.004],[-0.912,-0.868],[-14.191,-12.528],[-5.717,3.289],[0,0]],\\"v\\":[[44.596,42.757],[44.596,42.756],[45.846,48.332],[70.133,44.579],[85.636,25.503],[82.906,22.996],[49.163,8.976],[44.596,42.757]],\\"c\\":true},\\"hd\\":false}},{\\"ty\\":\\"sh\\",\\"d\\":1,\\"ks\\":{\\"a\\":0,\\"k\\":{\\"i\\":[[0,0],[-1.682,-3.173],[-1.569,4.998],[5.807,0.729],[-1.779,-3.056]],\\"o\\":[[1.808,3.104],[2.302,-5.357],[-5.225,-1.186],[1.909,2.977],[0,0]],\\"v\\":[[123.102,62.031],[128.337,71.447],[134.163,55.865],[117.568,52.981],[123.102,62.031]],\\"c\\":true},\\"hd\\":false}},{\\"ty\\":\\"sh\\",\\"d\\":1,\\"ks\\":{\\"a\\":0,\\"k\\":{\\"i\\":[[0,0],[3.536,-4.636],[-7.166,-0.337],[3.586,3.862]],\\"o\\":[[-3.642,3.907],[7.166,-0.333],[-3.532,-4.679],[0,0]],\\"v\\":[[91.087,31.027],[80.287,43.877],[101.797,43.882],[91.087,31.027]],\\"c\\":true},\\"hd\\":false}},{\\"ty\\":\\"sh\\",\\"d\\":1,\\"ks\\":{\\"a\\":0,\\"k\\":{\\"i\\":[[0,0],[-1.931,2.995],[5.193,-1.156],[-2.264,-5.339],[-1.785,3.079]],\\"o\\":[[1.788,-3.082],[-5.769,0.717],[1.582,5.069],[1.638,-3.159],[0,0]],\\"v\\":[[58.867,62.084],[64.446,52.967],[47.946,55.784],[53.73,71.442],[58.867,62.084]],\\"c\\":true},\\"hd\\":false}},{\\"ty\\":\\"sh\\",\\"d\\":1,\\"ks\\":{\\"a\\":0,\\"k\\":{\\"i\\":[[0,0],[1.83,3.142],[1.646,3.161],[1.607,-5.159],[-5.96,-0.661]],\\"o\\":[[-1.979,-3.052],[-1.793,-3.08],[-2.312,5.453],[5.191,1.124],[0,0]],\\"v\\":[[64.612,108.279],[58.898,98.986],[53.738,89.623],[47.839,105.598],[64.612,108.279]],\\"c\\":true},\\"hd\\":false}},{\\"ty\\":\\"sh\\",\\"d\\":1,\\"ks\\":{\\"a\\":0,\\"k\\":{\\"i\\":[[0,0],[-3.544,4.789],[3.71,0],[3.485,0.136],[-3.685,-3.954]],\\"o\\":[[3.562,-3.9],[-3.61,0.16],[-3.611,0],[3.59,4.703],[0,0]],\\"v\\":[[91.302,130.214],[101.987,117.146],[91,117.389],[80.352,117.184],[91.302,130.214]],\\"c\\":true},\\"hd\\":false}},{\\"ty\\":\\"sh\\",\\"d\\":1,\\"ks\\":{\\"a\\":0,\\"k\\":{\\"i\\":[[0,0],[1.831,-3.157],[1.923,-3.033],[-5.277,1.197],[2.432,5.513]],\\"o\\":[[-1.688,3.234],[-1.801,3.105],[6.003,-0.716],[-1.628,-5.072],[0,0]],\\"v\\":[[128.412,89.345],[123.133,98.933],[117.546,108.141],[134.523,105.263],[128.412,89.345]],\\"c\\":true},\\"hd\\":false}},{\\"ty\\":\\"sh\\",\\"d\\":1,\\"ks\\":{\\"a\\":0,\\"k\\":{\\"i\\":[[0,0],[0,0],[-2.431,4.973],[2.773,4.754],[3.087,4.536],[5.635,0],[5.413,-0.407],[2.763,-4.766],[0,0],[2.396,-4.958],[-2.773,-4.761],[-3.142,-4.563],[-5.663,0],[-5.4,0.383],[-2.757,4.762]],\\"o\\":[[0,0],[2.781,-4.789],[-2.456,-4.923],[-2.757,-4.742],[-5.411,-0.41],[-5.61,0],[-3.118,4.543],[0,0],[-2.764,4.764],[2.402,4.956],[2.786,4.786],[5.305,0.342],[5.687,0],[3.062,-4.574],[0,0]],\\"v\\":[[116.398,95.071],[116.399,95.071],[124.219,80.423],[116.373,65.904],[107.604,51.983],[91.001,51.358],[74.428,51.978],[65.604,65.946],[65.604,65.945],[57.862,80.533],[65.627,95.113],[74.522,109.142],[91,109.659],[107.667,109.078],[116.398,95.071]],\\"c\\":true},\\"hd\\":false}},{\\"ty\\":\\"sh\\",\\"d\\":1,\\"ks\\":{\\"a\\":0,\\"k\\":{\\"i\\":[[0,0],[13.437,-11.627],[1.363,-1.306],[-5.002,-7.08],[-7.55,-1.732],[-0.251,1.204],[6.094,3.495]],\\"o\\":[[-5.722,-3.281],[-1.426,1.237],[5.202,5.572],[8.584,0.801],[0.295,-1.194],[3.829,-18.481],[0,0]],\\"v\\":[[132.437,8.746],[100.72,21.682],[96.535,25.499],[111.92,44.589],[136.254,48.412],[137.073,44.814],[132.437,8.746]],\\"c\\":true},\\"hd\\":false}},{\\"ty\\":\\"sh\\",\\"d\\":1,\\"ks\\":{\\"a\\":0,\\"k\\":{\\"i\\":[[0,0],[0,-8.924],[8.976,0],[0,8.925],[-8.975,0]],\\"o\\":[[8.976,0],[0,8.925],[-8.975,0],[0,-8.924],[0,0]],\\"v\\":[[91,64.188],[107.252,80.347],[91,96.507],[74.748,80.347],[91,64.188]],\\"c\\":true},\\"hd\\":false}},{\\"ty\\":\\"fl\\",\\"c\\":{\\"a\\":0,\\"k\\":[0.325,0.757,0.871,1]},\\"hd\\":false,\\"o\\":{\\"a\\":0,\\"k\\":100},\\"r\\":2},{\\"ty\\":\\"tr\\",\\"p\\":{\\"a\\":0,\\"k\\":[128,133.5]},\\"a\\":{\\"a\\":0,\\"k\\":[91,80.5]},\\"s\\":{\\"a\\":0,\\"k\\":[100,100]},\\"r\\":{\\"a\\":1,\\"k\\":[{\\"t\\":0,\\"s\\":[0],\\"i\\":{\\"x\\":[0.833],\\"y\\":[0.833]},\\"o\\":{\\"x\\":[0.167],\\"y\\":[0.167]},\\"e\\":[360]},{\\"t\\":200,\\"s\\":[360]}]},\\"o\\":{\\"a\\":0,\\"k\\":100},\\"sk\\":{\\"a\\":0,\\"k\\":0},\\"sa\\":{\\"a\\":0,\\"k\\":0}}],\\"nm\\":\\"reactlogoyo\\",\\"ln\\":\\"reactlogoyo\\",\\"hd\\":false},{\\"ty\\":\\"gr\\",\\"it\\":[{\\"d\\":1,\\"ty\\":\\"el\\",\\"s\\":{\\"a\\":0,\\"k\\":[256,256]},\\"p\\":{\\"a\\":0,\\"k\\":[0,0]}},{\\"ty\\":\\"fl\\",\\"c\\":{\\"a\\":0,\\"k\\":[0.133,0.133,0.133,1]},\\"hd\\":false,\\"o\\":{\\"a\\":0,\\"k\\":100},\\"r\\":2},{\\"ty\\":\\"tr\\",\\"p\\":{\\"a\\":0,\\"k\\":[0,0]},\\"a\\":{\\"a\\":0,\\"k\\":[-128,-128]},\\"s\\":{\\"a\\":0,\\"k\\":[100,100]},\\"r\\":{\\"a\\":0,\\"k\\":0},\\"o\\":{\\"a\\":0,\\"k\\":100},\\"sk\\":{\\"a\\":0,\\"k\\":0},\\"sa\\":{\\"a\\":0,\\"k\\":0}}],\\"nm\\":\\"Object\\",\\"hd\\":false}]}]}"
        speed={1}
        style={
          Object {
            "aspectRatio": 1,
            "height": "100%",
            "width": 50,
          }
        }
      />
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "fontSize": 21,
          "fontWeight": "bold",
        },
        Object {
          "color": "#000000",
        },
      ]
    }
  >
    You are successfully registered.
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontSize": 10,
          "marginBottom": 10,
        },
        Object {
          "color": "#000000",
        },
      ]
    }
  >
    Thank you for registering. To confirm your email address, please check your email and click the link we sent. If you cannot check in the inbox, then make sure you check your spam folder as well.
  </Text>
</RNCSafeAreaView>
`;
